backend/inspirehep/orcid/tasks.py

```diff
@@ -260,6 +262,10 @@ def orcid_push(self, orcid, rec_id, oauth_token, kwargs_to_pusher=None):
         # thus backoff power 5 is (secs): [5, 25, 125, 625]
         backoff = 5 ** (self.request.retries + 1)
         raise self.retry(max_retries=4, countdown=backoff, exc=exc)
+    except MovedPermanentlyException as exc:
+        old_orcid, new_orcid = re.findall(ORCID_REGEX, exc.args[0])
+        update_moved_orcid(old_orcid, new_orcid)
```

Comment:
wouldn't it be better to move this kind of business logic to the `domain_models` `push` function?